generate.py

```diff
@@ -724,3 +733,185 @@ def normalize(code):
+---
+{content}
+''')
+
+
+### Generate WMT event files
+for event in WMT_EVENTS:
+
+  name = event['name']
+
+  # Start and end dates are necessary for listing the events on events.md
+  if not isinstance(event['startDate'], str):
+    raise Exception(event['startDate'])
+
+  if not isinstance(event['endDate'], str):
+    raise Exception(event['endDate'])
+
+  if event.get('blockquotes', None):
+    if not isinstance(event['blockquotes'], list):
+      raise Exception(event['blockquotes'])
+
+  if event.get('urlsAfterBlockquote', None):
+    if not isinstance(event['urlsAfterBlockquote'], list):
+      raise Exception(event['urlsAfterBlockquote'])
+
+  if event.get('impDatesHeaderH4', None):
+    if not isinstance(event['impDatesHeaderH4'], list):
+      raise Exception(event['impDatesHeaderH4'])
+
+  if event.get('stBulletTasks', None):
+    if not isinstance(event['stBulletTasks'], list):
+      raise Exception(event['stBulletTasks'])
+
+  if event.get('gmtList', None):
+    if not isinstance(event['gmtList'], list):
+      raise Exception(event['gmtList'])
+
+  if event.get('lpBulletList', None):
+    if not isinstance(event['lpBulletList'], list):
+      raise Exception(event['lpBulletList'])
+
+  if event.get('spBulletTopics', None):
+    if not isinstance(event['spBulletTopics'], list):
+      raise Exception(event['spBulletTopics'])
+
+  if event.get('rpBlockquotes', None):
+    if not isinstance(event['rpBlockquotes'], list):
+      raise Exception(event['rpBlockquotes'])
+
+  if event.get('spBlockquotes', None):
+    if not isinstance(event['spBlockquotes'], list):
+      raise Exception(event['spBlockquotes'])
+
+  if event.get('pfTitles', None):
+    if not isinstance(event['pfTitles'], list):
+      raise Exception(event['pfTitles'])
+
+  if event.get('psBulletTitles', None):
+    if not isinstance(event['psBulletTitles'], list):
+      raise Exception(event['psBulletTitles'])
+
+  if event.get('scheduleDays', None):
+    if not isinstance(event['scheduleDays'], list):
+      raise Exception(event['scheduleDays'])
+
+  if event.get('languages', None):
+    if not isinstance(event['languages'], list):
+      raise Exception(event['languages'])
+
+  if event.get('cfpContent', None):
+    if not isinstance(event['cfpContent'], list):
+      raise Exception(event['cfpContent'])
+
+  if event.get('cfwWorkshopsBlockquotes', None):
+    if not isinstance(event['cfwWorkshopsBlockquotes'], list):
+      raise Exception(event['cfwWorkshopsBlockquotes'])
+
+  if event.get('tutorialsBlockquotoes', None):
+    if not isinstance(event['tutorialsBlockquotoes'], list):
+      raise Exception(event['tutorialsBlockquotoes'])
+
+  frontmatter = {
+    'autogenerated': True,
+    'autogenerated_from': 'wmt_events.json',
+    'parent': 'Events',
+    'nav_order': event['startDate'].split('-')[0],
+    'layout': 'wmt_event',
+    **event
+  }
+
+  # Doesn't create a file if it's an external event, as external events should not have an id
```

Comment:
What's an example of an "external" event?

Reply:
For example, ```AmericasNLP```. We don't have a page for it but we link the event externally.